lbz/authz.py

```diff
@@ -166,31 +166,28 @@ def sign_authz(authz_data: dict, private_key_jwk: dict) -> str:
         )
 
 
-def check_permission(resource: Resource, permission_name: str) -> dict:
+def check_permission(resource: Union[Type[Resource], Resource], permission_name: str) -> dict:
     """
     Check if requester has sufficient permissions to do something on specific resource.
 
     Raises if not.
     """
-    authorization_header = resource.request.headers.get("Authorization")
-    authorization_scope = None
-    if not authorization_header:
-        if hasattr(resource, "get_guest_authorization"):
-            authorization_scope = resource.get_guest_authorization()
-        else:
-            raise Unauthorized("Authorization header missing or empty")
+    authorization_header = resource.request.headers.get("Authorization", "")
+    authorization_scope = resource.get_guest_authorization()
```

Comment:
I think `guest_authorization_policy` instead of `authorization_scope` would be less confusing

What if the resource does not have this method implemented?

Reply:
1. make sense.
2. not an issue. check below.